Source/player.cpp

```diff
@@ -1143,16 +1143,15 @@ bool PlrHitPlr(int pnum, int8_t p)
 
 bool PlrHitObj(int pnum, int mx, int my)
 {
-	int oi;
-
-	if (dObject[mx][my] > 0) {
-		oi = dObject[mx][my] - 1;
-	} else {
-		oi = -dObject[mx][my] - 1;
+	int oi = abs(dObject[mx][my]) - 1;
+	if (oi < 0) {
+		return false;
 	}
 
-	if (Objects[oi]._oBreak == 1) {
-		BreakObject(pnum, oi);
+	Object &targetObject = Objects[oi];

```

Comment:
This only get called if the position is known to have a positive id. Might as well have it take `Object &targetObject` and make the call `PlrHitObj(pnum, Objects[dObject[dx][dy] - 1])`